warehouse/integrations/deltalake/deltalake.go

```diff
@@ -500,164 +552,346 @@ func getTableSchemaDiff(tableSchemaInUpload, tableSchemaAfterUpload model.TableS
 }
 
 // loadTable Loads table with table name
-func (dl *Deltalake) loadTable(tableName string, tableSchemaInUpload, tableSchemaAfterUpload model.TableSchema, skipTempTableDelete bool) (stagingTableName string, err error) {
-	// Getting sorted column keys from tableSchemaInUpload
-	sortedColumnKeys := warehouseutils.SortColumnKeysFromColumnMap(tableSchemaInUpload)
+func (d *Deltalake) loadTable(tableName string, tableSchemaInUpload, tableSchemaAfterUpload model.TableSchema, skipTempTableDelete bool) (string, error) {
+	var (
+		sortedColumnKeys = warehouseutils.SortColumnKeysFromColumnMap(tableSchemaInUpload)
+		stagingTableName = warehouseutils.StagingTableName(provider, tableName, tableNameLimit)
 
-	// Creating staging table
-	stagingTableName = warehouseutils.StagingTableName(provider, tableName, tableNameLimit)
-	err = dl.CreateTable(stagingTableName, tableSchemaAfterUpload)
-	if err != nil {
-		return
+		err       error
+		auth      string
+		startedAt time.Time
+	)
+
+	d.Logger.Infow("started loading",
+		logfield.SourceID, d.Warehouse.Source.ID,
+		logfield.SourceType, d.Warehouse.Source.SourceDefinition.Name,
+		logfield.DestinationID, d.Warehouse.Destination.ID,
+		logfield.DestinationType, d.Warehouse.Destination.DestinationDefinition.Name,
+		logfield.WorkspaceID, d.Warehouse.WorkspaceID,
+		logfield.Namespace, d.Namespace,
+		logfield.TableName, tableName,
+	)
+
+	if err = d.CreateTable(stagingTableName, tableSchemaAfterUpload); err != nil {
+		d.Logger.Warnw("failure creating temporary table",
+			logfield.SourceID, d.Warehouse.Source.ID,
+			logfield.SourceType, d.Warehouse.Source.SourceDefinition.Name,
+			logfield.DestinationID, d.Warehouse.Destination.ID,
+			logfield.DestinationType, d.Warehouse.Destination.DestinationDefinition.Name,
+			logfield.WorkspaceID, d.Warehouse.WorkspaceID,
+			logfield.Namespace, d.Namespace,
+			logfield.TableName, tableName,
+			logfield.StagingTableName, stagingTableName,
+			logfield.Error, err.Error(),
+		)
+
+		return "", fmt.Errorf("creating staging table: %w", err)
```

Comment:
How about logging upstream only as ERROR instead of warning? Is the logger really necessary here? If yes then I guess it should be ERROR. Wdyt?